### ISSUE TITLE
False positive for FastAPI return types.

### ISSUE THREAD

**Issue by danpker** [snok/flake8-type-checking#185]
As mentioned in #52 

According to this test, `CustomType` should be moved to a `TYPE_CHECKING` block.

```python
@pytest.mark.parametrize('fdef', ['def', 'async def'])
def test_api_router_decorated_function_return_type(fdef):
    """
    We don't care about return types. To my knowledge,
    these are not evaluated by FastAPI/pydantic.
    """
    example = textwrap.dedent(f'''
        from fastapi import APIRouter
        from fastapi import Request

        from app.types import CustomType

        some_router = APIRouter(prefix='/some-path')

        @some_router.get('/{{resource_id}}')
        {fdef} list_something(request: Request) -> CustomType:
            return None
        ''')
    assert _get_error(example, error_code_filter='TC001,TC002,TC003', **defaults) == {
        '5:0 ' + TC002.format(module='app.types.CustomType')
    }
```

However, doing so, even with a pretty basic route gives the following error in FastAPI:

```python
     def test() -> Test:
                   ^^^^
 NameError: name 'Test' is not defined
```

**Comment by sondrelg** [comment]
Agree that this seems unexpected. Looking at this quickly, I would expect:

```python
from fastapi import APIRouter
from fastapi import Request

from app.types import CustomType

some_router = APIRouter(prefix='/some-path')


@some_router.get('/{{resource_id}}')
def list_something(request: Request) -> "CustomType":
    return None
```

or 

```python
from __future__ import annotations

from fastapi import APIRouter
from fastapi import Request

from app.types import CustomType

some_router = APIRouter(prefix='/some-path')


@some_router.get('/{{resource_id}}')
def list_something(request: Request) -> CustomType:
    return None
```

To be correct, wrt. to the return type.

Would you be interested in creating a fix for this @danpker? I'm on holiday at the moment, so won't have time to look at this, but will make time to push a fix if one is created 👍
